chain-signatures/node/src/protocol/triple.rs

```diff
@@ -325,10 +674,11 @@ impl TripleManager {
             if self.len_potential().await >= cfg.triple.max_triples as usize {
                 false
             } else {
+                let tasks = self.tasks.read().await;
                 // We will always try to generate a new triple if we have less than the minimum
                 self.len_mine().await < cfg.triple.min_triples as usize
-                    && self.introduced.len() < cfg.max_concurrent_introduction as usize
-                    && self.generators.len() < cfg.max_concurrent_generation as usize
+                    && tasks.introduced.len() < cfg.max_concurrent_introduction as usize
```

Comment:
is this same as self.len_introduced()?

Reply:
yes, but the reason why we don't use it is because we'd be acquiring another read lock which we've already acquired a write lock so no point in calling the function directly